src/common/bit_vector.hpp

```diff
@@ -372,11 +372,8 @@ void BitVector<_b>::decode_data(bufferlist::const_iterator& it,
   while (data_byte_offset < end_offset) {
     uint64_t len = std::min<uint64_t>(BLOCK_SIZE, end_offset - data_byte_offset);
 
-    bufferptr ptr;
-    it.copy_deep(len, ptr);
-
     bufferlist bit;
-    bit.append(ptr);
+    bit.append(it.copy_deep(len));
```

Comment:
would it be faster to add an append_from_deep_copy() interface to the bufferlist? or is the code created similar?